lldb/test/API/lang/swift/async/unwind/unwind_in_all_instructions/main.swift

```diff
@@ -0,0 +1,49 @@
+func async_work() async {
+  print("async working")
+}
+func work() {
+  print("working")
+}
+
+func ASYNC___1___(cond: Int) async -> Int {
+  work()
+  await async_work()
+  for i in 1...3 {
+    work()
+    await async_work()
+    work()
+    if (cond + i == 3) {
+      await async_work()
+      print("breaking here!")
+      break;
+    }
+  }
+  return 0
+}
+
+func ASYNC___2___() async -> Int {
+  let result = await ASYNC___1___(cond: 1)
+  return result
+}
+
+func ASYNC___3___() async -> Int {
+  let result = await ASYNC___2___()
+  return result
+}
+
+func ASYNC___4___() async -> Int {
+  let result = await ASYNC___3___()
+  return result
+}
+
+func ASYNC___5___() async -> Int {
+  let result = await ASYNC___4___()
+  return result
+}
```

Comment:
does this test need 5 layers of async indirection? If so, can you document the reason why. If not, can you reduce it to however many are needed. If there's not a concrete reason, but a preference, can you document that :)

Reply:
Yup! There is a good reason, let me write it down as a comment in the test!